java/integTest/src/test/java/glide/modules/JsonTests.java

```diff
@@ -157,6 +159,58 @@ public void json_set_get_formatting() {
         assertEquals(expectedGetResult2, actualGetResult2);
     }
 
+    @Test
+    @SneakyThrows
+    public void arrappend() {
+        String key = UUID.randomUUID().toString();
+        String doc = "{\"a\": 1, \"b\": [\"one\", \"two\"]}";
+
+        assertEquals(OK, Json.set(client, key, "$", doc).get());
+
+        assertArrayEquals(
+                new Object[] {3L},
+                (Object[]) Json.arrappend(client, key, "$.b", new String[] {"\"three\""}).get());
+        assertEquals(
+                5L, Json.arrappend(client, key, ".b", new String[] {"\"four\"", "\"five\""}).get());
+
+        String getResult = Json.get(client, key, new String[] {"$"}).get();
+        String expectedGetResult =
+                "[{\"a\": 1, \"b\": [\"one\", \"two\", \"three\", \"four\", \"five\"]}]";
+        assertEquals(JsonParser.parseString(expectedGetResult), JsonParser.parseString(getResult));
+
+        assertArrayEquals(
+                new Object[] {null},
+                (Object[]) Json.arrappend(client, key, "$.a", new String[] {"\"value\""}).get());
+
+        // JSONPath, path doesn't exist
+        assertArrayEquals(
+                new Object[] {},
+                (Object[]) Json.arrappend(client, key, "$.c", new String[] {"\"value\""}).get());
+
+        // Legacy path, path doesn't exist
+        var exception =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> Json.arrappend(client, key, ".c", new String[] {"\"value\""}).get());
+
+        // Legacy path, the JSON value at path is not a array
+        exception =
+                assertThrows(
+                        ExecutionException.class,
+                        () -> Json.arrappend(client, key, ".a", new String[] {"\"value\""}).get());
+
+        exception =
+                assertThrows(
+                        ExecutionException.class,
+                        () ->
+                                Json.arrappend(client, "non_existing_key", "$.b", new String[] {"\"six\""}).get());
```

Comment:
nit: Spacing/indentation is inconsistent here with the other cases

Reply:
It is the result of `spotlessApply`.